klt-cert-manager/controllers/keptnwebhookcontroller/webhook_cert_controller_test.go

```diff
@@ -0,0 +1,364 @@
+		Status: apiv1.CustomResourceDefinitionStatus{},
+	}
+
+	crd2 := &apiv1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "crd2",
+		},
+		Spec: apiv1.CustomResourceDefinitionSpec{
+			Group: "Someonelese",
+			Conversion: &apiv1.CustomResourceConversion{
+				Strategy: strategyWebhook,
+				Webhook: &apiv1.WebhookConversion{
+					ClientConfig: &apiv1.WebhookClientConfig{
+						CABundle: []byte("my unmodified bundle"),
+					},
+				},
+			},
+		},
+	}
+	crd3 := &apiv1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "crd3",
+			Labels: map[string]string{
+				"crdGroup": "lifecycle.keptn.sh",
+			},
+		},
+		Spec: apiv1.CustomResourceDefinitionSpec{
+			Group: crdGroup,
+		},
+	}
+
+	t.Run(`reconcile successfully with mutatingwebhookconfiguration`, func(t *testing.T) {
+		fakeClient := fake.NewClient(crd1, crd2, crd3, &admissionregistrationv1.MutatingWebhookConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: Webhookconfig,
+			},
+			Webhooks: []admissionregistrationv1.MutatingWebhook{
+				{
+					ClientConfig: admissionregistrationv1.WebhookClientConfig{},
+				},
+				{
+					ClientConfig: admissionregistrationv1.WebhookClientConfig{},
+				},
+			},
+		})
+
+		controller, request := prepareController(t, fakeClient)
+		result, err := controller.Reconcile(context.TODO(), request)
+
+		assert.NoError(t, err)
+		assert.NotNil(t, result)
+	})
+	t.Run(`update crd successfully with up-to-date secret`, func(t *testing.T) {
+		fakeClient := fake.NewClient(crd1, crd2, crd3)
+		cs := newCertificateSecret()
+		_ = cs.setSecretFromReader(context.TODO(), fakeClient, namespace, testr.New(t))
+		_ = cs.validateCertificates(namespace)
+		_ = cs.createOrUpdateIfNecessary(context.TODO(), fakeClient)
+
+		controller, request := prepareController(t, fakeClient)
+		result, err := controller.Reconcile(context.TODO(), request)
+		require.NoError(t, err)
+		assert.NotNil(t, result)
+
+		expectedBundle, err := cs.loadCombinedBundle()
+		require.NoError(t, err)
+		actualCrd := &apiv1.CustomResourceDefinition{}
+
+		// crd1 should get a new secret
+		err = fakeClient.Get(context.TODO(), client.ObjectKey{Name: crd1.Name}, actualCrd)
+		require.NoError(t, err)
+		assert.Equal(t, expectedBundle, actualCrd.Spec.Conversion.Webhook.ClientConfig.CABundle)
+
+		// crd2 is not a keptn resource and should not be touched
+		err = fakeClient.Get(context.TODO(), client.ObjectKey{Name: crd2.Name}, actualCrd)
+		require.NoError(t, err)
+		assert.Equal(t, crd2.Spec.Conversion.Webhook.ClientConfig.CABundle, actualCrd.Spec.Conversion.Webhook.ClientConfig.CABundle)
+
+		// crd 3 should not have a webhook conversion
+		err = fakeClient.Get(context.TODO(), client.ObjectKey{Name: crd3.Name}, actualCrd)
+		require.NoError(t, err)
+		assert.Empty(t, actualCrd.Spec.Conversion.Webhook)
+	})
+	// Generation must not be skipped because webhook startup routine listens for the secret
+	// See cmd/operator/manager.go and cmd/operator/watcher.go
```

Comment:
those two files that this comment is referring to seem to not be present. Or will they be added later?

Reply:
they are added in the next pr yes :) I will put the proper path tho since it changed